addons/mass_mailing/models/mailing.py

```diff
@@ -1306,36 +1312,75 @@ def _unsubscribe_token(self, res_id, email):
         token = (self.env.cr.dbname, self.id, int(res_id), tools.ustr(email))
         return hmac.new(secret.encode('utf-8'), repr(token).encode('utf-8'), hashlib.sha512).hexdigest()
 
+
+    def _image_to_url(self, b64image: bytes):
+        """Store an image in an attachement and returns an url"""
+        attachment = self.env['ir.attachment'].create({
+            'datas': b64image,
+            'name': "cropped_image_mailing_{}".format(self.id),
+            'type': 'binary',})
+
+        attachment.generate_access_token()
+
+        return '/web/image/%s?access_token=%s' % (
+            attachment.id, attachment.access_token)
+
     def _convert_inline_images_to_urls(self, body_html):
         """
         Find inline base64 encoded images, make an attachement out of
         them and replace the inline image with an url to the attachement.
         """
-
-        def _image_to_url(b64image: bytes):
-            """Store an image in an attachement and returns an url"""
-            attachment = self.env['ir.attachment'].create({
-                'datas': b64image,
-                'name': "cropped_image_mailing_{}".format(self.id),
-                'type': 'binary',})
-
-            attachment.generate_access_token()
-
-            return '/web/image/%s?access_token=%s' % (
-                attachment.id, attachment.access_token)
-
-        modified = False
         root = lxml.html.fromstring(body_html)
+        modified = False
         for node in root.iter('img'):
             match = image_re.match(node.attrib.get('src', ''))
             if match:
                 mime = match.group(1)  # unsed
                 image = match.group(2).encode()  # base64 image as bytes
 
-                node.attrib['src'] = _image_to_url(image)
+                node.attrib['src'] = self._image_to_url(image)
                 modified = True
 
         if modified:
             return lxml.html.tostring(root, encoding='unicode')
 
         return body_html
+
+    def _crop_vml_images(self, body_html):
+        """
+        Find VML v:image elements, crop their source images, make an attachement
+        out of them and replace their source with an url to the attachement.
+        """
+        root = lxml.html.fromstring(body_html)
+        modified = False
+        for comment in root.xpath('//comment()'):
+            is_mso = mso_re.match(comment.text)
+            if is_mso:
+                for match in re.findall(r'<v:image[^>]*>', comment.text):
+                    try:
+                        url = re.search(r'src=\s*\"([^\"]+)\"', match).group(1)
+                        absolute_url = url
+                        if url.startswith('/'):
+                            absolute_url = self.get_base_url() + url
+                        request = requests.get(absolute_url)
```

Comment:
@odoo/rd-security The alert gets triggered because of this line. I need to do this because we're cropping an image from an url retrieved from the `src` attribute of an element. Before cropping it I need to actually fetch the image. I don't know what kind of attack could be done this way since if what we requested turns out not to be an image, `tools.ImageProcess` will fail with an `UnidentifiedImageError`. Please advise :-)

Reply:
@Zinston it's always risky to get unknown URL. What if I spam 100 calls with <https://releases.ubuntu.com/22.04.2/ubuntu-22.04.2-desktop-amd64.iso>? You should have a small timeout and check the size of what you are going to download first. You can use `requests.head` to verify those.
Also did not test but it seems the request is made before we verify the access to the model (before the `super`). So even a portal without access to the model could force you fetching arbitrary URLs.